src/Core/functions.php

```diff
@@ -249,6 +249,31 @@ function env($key, $default = null)
 
 }
 
+if (!function_exists('warning')) {
```

Comment:
``warning()`` seems quite general. There is a good chance of a name clash with user land code. How about ``throwWarning()`` instead?

Reply:
Why does this need to exist in the global space anyways?
Additionally, "throw" is not quite what it is supposed to do, logWarning or maybe processWarning or even cacheWarning, as it could live in the cache namespace anyways and be included - maybe a trait - from the cache engines only, instead of poluting global space :P

Reply:
http://php.net/manual/en/function.trigger-error.php
- `triggerWarning()`
- `generateWarning()`

I can for sure move this to `class CacheEngine` if any of the core members prefer that as well.

Reply:
Yeah, sorry, I agree with @hmic on the name. I just converted what the doc block stated into its name. 😼 
``triggerWarning()`` sounds good to me.